modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/add_params.ts

```diff
@@ -14,12 +14,8 @@
 
 /**
  * Appends given parameters to the given URL.
- * @param {String | Object} params Parameters to be added on to the base url
- * @param {String} baseUrl Base url
- * @return {String} url URL constructed from base url and params
- * @review
  */
-export default function addParams(params, baseUrl) {
+export default function addParams(params: string | object, baseUrl: string) {
```

Comment:
We most certainly don't want `object` here but something like `{ [key:string]: string }`... does that have a universal name?

Also... see comments about runtime type guards in the commits... keep'em or drop'em?

Additional future warning... `typeof params !== 'object'` might not be telling you what you thing is telling you... maybe we should lint about this or something, idk!

Reply:
> `{ [key:string]: string }`... does that have a universal name ?

Nope, but we can create one if it is very common... 

I usually create types for this kind of option arguments. So something like this I think it's possible:

```
addParamsOptions {
   knownOption1: string;
   knownOption2: boolean;
   [key:string]: string;
}
```

If there are no known options just the `[key:string]: string;` part. But having it as a named (and exported) type is usually handy when you need to pass those options through several function calls and have them typed.

Of course I'm not saying 100% of the time you must do this, but if it is a public API there are many benefits in having a named type for this, IMO.

Reply:
If we don't have known properties on the object, instead of using `[key:string]: string;` we have the `Record` that was introduced in TypeScript, we can replace it with `Record<string, string>`;

Reply:
> we can replace it with Record<string, string>
> +1

Nice! Completely missed that! Works like a charm! ⭐ 

> I prefer we don't create a global [key:string]: string; helper, as I think it may lead people to abuse it when they don't want to define a proper interface.

This is likely to happen anyways, isn't it? They can even use `any` 😂 . But yeah, totally get your point, let's keep those to a minimum until we know what's really useful and valuable! 👍